### PR TITLE
added angular momentum and linear momentum as built in functions

### PR DESCRIPTION
**Pull request by EveCharbie**
Still need an example

### CODE REVIEW

File: bioptim/limits/penalty.py [hunk 1]
```diff
@@ -404,6 +404,90 @@ def minimize_com_velocity(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes:
             com_dot_cx = BiorbdInterface.mx_to_cx("com_dot", nlp.model.CoMdot, nlp.states["q"], nlp.states["qdot"])
             return com_dot_cx
 
+        @staticmethod
+        def minimize_com_acceleration(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes: Union[tuple, list] = None):
```
Comment (pariterre) [inline bioptim/limits/penalty.py:408]: Please add a test for this penalty

File: bioptim/limits/penalty.py [hunk 2]
```diff
@@ -404,6 +404,90 @@ def minimize_com_velocity(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes:
...
+                )
+            qddot = nlp.states["qddot"] if "qddot" in nlp.states.keys() else nlp.controls["qddot"]
+
+            com_ddot_cx = BiorbdInterface.mx_to_cx(
+                "com_ddot", nlp.model.CoMddot, nlp.states["q"], nlp.states["qdot"], qddot
+            )
+            return com_ddot_cx
+
+        @staticmethod
+        def minimize_angular_momentum(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes: Union[tuple, list] = None):
```
Comment (pariterre) [inline bioptim/limits/penalty.py:441]: Please add a test for this penalty

File: bioptim/limits/penalty.py [hunk 3]
```diff
@@ -404,6 +404,90 @@ def minimize_com_velocity(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes:
...
+            PenaltyFunctionAbstract.set_axes_rows(penalty, axes)
+            penalty.quadratic = True if penalty.quadratic is None else penalty.quadratic
+            nlp = all_pn.nlp
+            angular_momentum_cx = BiorbdInterface.mx_to_cx(
+                "angular_momentum", nlp.model.angularMomentum, nlp.states["q"], nlp.states["qdot"]
+            )
+            return angular_momentum_cx
+
+        @staticmethod
+        def minimize_linear_momentum(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes: Union[tuple, list] = None):
```
Comment (pariterre) [inline bioptim/limits/penalty.py:465]: Please add a test for this penalty

**Review by pariterre — COMMENTED** [review]
Apart from the tests, all seems good :)

**Comment by Ipuch** [comment]
@pariterre mergeable :+1:

File: bioptim/limits/penalty.py [hunk 4]
```diff
@@ -404,6 +404,95 @@ def minimize_com_velocity(penalty: PenaltyOption, all_pn: PenaltyNodeList, axes:
...
+
+            nlp = all_pn.nlp
+            com_velocity = BiorbdInterface.mx_to_cx(
+                "com_velocity", nlp.model.CoMdot, nlp.states["q"], nlp.states["qdot"]
+            )
+            if isinstance(com_velocity, SX):
+                mass = Function("mass", [], [nlp.model.mass().to_mx()]).expand()
+                mass = mass()["o0"]
+            else:
+                mass = nlp.model.mass().to_mx()
```
Comment (pariterre) [inline bioptim/limits/penalty.py:492]: Is this else useful? If yes, please test it, otherwise it should be removed

File: tests/test_penalty.py [hunk 5]
```diff
@@ -55,9 +57,13 @@ def prepare_test_ocp(with_muscles=False, with_contact=False, with_actuator=False
         nx = biorbd_model.nbQ() + biorbd_model.nbQdot()
         nu = biorbd_model.nbGeneralizedTorque()
     x_init = InitialGuess(np.zeros((nx, 1)))
-    u_init = InitialGuess(np.zeros((nu, 1)))
+    if implicit:
+        mod = 2
+    else:
+        mod = 1
```
Comment (pariterre) [inline tests/test_penalty.py:63]: Please use one line if statement for these kind of small tests: 
`mod = 2 if implicit else 1`

**Review by pariterre — CHANGES_REQUESTED** [review]
Almost there!

File: bioptim/limits/penalty.py (hunk 4 above)
Comment (Ipuch) [reply bioptim/limits/penalty.py:492]: Yes, it is. If one uses Sx or MX it doesn't work the same way.

**Review by pariterre — CHANGES_REQUESTED** [review]

**Comment by Ipuch** [comment]
https://github.com/pyomeca/bioptim/pull/437#discussion_r821978024
It didn't work with the interface, I don't know why...

**Review by pariterre — APPROVED** [review]

**Pull request merged**